drv/stm32h7-sprot-server/src/main.rs

```diff
@@ -124,12 +129,24 @@ cfg_if::cfg_if! {
     }
 }
 
+/// Return an error if the expected MsgType doesn't match the actual MsgType
+macro_rules! expect_msg {
+    ($expected:expr, $actual:expr) => {
+        if $expected != $actual {
+            ringbuf_entry!(Trace::WrongMsgType($actual));
+            Err(SprotError::BadMessageType)
+        } else {
+            Ok(())
+        }
+    };
+}
+
 pub struct ServerImpl {
     sys: sys_api::Sys,
     spi: drv_spi_api::SpiDevice,
     // Use separate buffers so that retries can be generic.
-    pub tx_buf: [u8; BUF_SIZE],
-    pub rx_buf: [u8; BUF_SIZE],
+    pub tx_buf: TxMsg,
```

Comment:
Nit: maybe name these `tx/rx_msg`, since they're not bare buffers?

(same thing below)

Reply:
They might go back to being bare buffers :)